File: codelearn/sasuke30/d/d.py
from collections import defaultdict

def myfn (c):
    return [-c[0], c[1]]

def chooseEmployee(skill, salary, requestSkill, k):
    a = defaultdict(int)
    p = defaultdict(list)

    for i in range(len(skill)):
        p[skill[i]].append(salary[i])
    for c in p: 
        sorted(p[c])

    for c in skill: 
        a[c] += 1

    b = list()
    for c in a: 
        b.append((a[c], c))

    b.sort(key=myfn)

    res = 0 
    y = p[requestSkill]
    y.sort()
    for c in y: 
        if k > 0: 
            res += c 
            k -= 1

    if k > 0: 
        for c in b : 
            y = p[c[1]]
            y.sort()
            if c[1] != requestSkill: 
                for z in y: 
                    if k > 0: 
                        res += z
                        k -= 1 

    if k > 0: 
        return -1 
    else:
        return res

if __name__ == "__main__":
    print(chooseEmployee(["Java",".Net","C++","Java"], [20,40,60,80], ".Net", 3))
    print(chooseEmployee(["B","C","C","A","B","B","A","A"], [30, 50, 20, 10, 40, 60, 80, 70], "C", 4))

File: codelearn/sasuke30/d/d.cpp
using namespace std;
#include<bits/stdc++.h>
#define Debug(...) fprintf(stderr, __VA_ARGS__)
#define st first
#define nd second
#define FORE(i,a,b) for(int i=(a),b_=(b);i<=b_;++i)
#define FORD(i,a,b) for(int i=(a),b_=(b);i>=b_;--i)
#define TR(c, it) for(__typeof((c).begin()) it=(c).begin(); it!=(c).end(); it++)
#define removeDuplicate(a) sort(a.begin(), a.end()); a.resize(distance(a.begin(), unique(a.begin(), a.end())))
 
const int N = 2e5 + 1;
const int MOD = 1e9 + 7; 

#define st first
#define nd second

const int N = (int) 1e5 + 5;

pair <int, pair <string, int> > x[N];

int chooseEmployee(std::vector<std::string> skill, std::vector<int> salary, std::string requestSkill, int k)
{
    map <string, int> cnt;
    int n = 0;
    int choose = 0;
    int ans = 0;
    if (choose >= k) return ans;
    for (int i = 0; i < skill.size(); i++) {
       cnt[skill[i]]++;
    }
    for (int i = 0; i < skill.size(); i++) x[i] = make_pair(cnt[skill[i]], make_pair(skill[i], -salary[i]));
    sort(x, x + (int) skill.size());

    int j = -1;
    for (int i = skill.size() - 1; i >= 0; i--)
      if (x[i].nd.st == requestSkill) {
        j = i;
        break;
      }
    if (j != -1) {
      for (int i = j; i >= 0; i--) {
        if (x[i].nd.st == requestSkill) {
          choose++;
          ans += -x[i].nd.nd;
        } else break;
        if (choose >= k) break;
      }
    }
    if (choose >= k) return ans;
    for (int i = skill.size() - 1; i >= 0; i--) {
      if (x[i].nd.st == requestSkill) continue;
      choose++;
      ans += -x[i].nd.nd;
      if (choose >= k) break;
    }
    if (choose < k) return -1;
    return ans;
}
 
int main() {
  #define TASK "TASK"
  // freopen(TASK".inp","r",stdin);
  // freopen(TASK".out","w",stdout);
  ios_base::sync_with_stdio(false); cin.tie(NULL);
     
  return 0;
}